src/tbp/monty/frameworks/environments/embodied_data.py

```diff
@@ -576,6 +591,51 @@ def get_good_view(
         TODO M : move most of this to the motor systems, shouldn't be in embodied_data
             class
         """
+        if self._use_get_good_view_positioning_procedure:
+            _configured_policy = self.motor_system._policy
+
+            self.motor_system._policy = GetGoodView(
+                agent_id=self.motor_system._policy.agent_id,
+                desired_object_distance=_configured_policy.desired_object_distance,
+                good_view_percentage=_configured_policy.good_view_percentage,
+                multiple_objects_present=self.num_distractors > 0,
+                sensor_id=sensor_id,
+                target_semantic_id=self.primary_target["semantic_id"],
+                allow_translation=allow_translation,
+                max_orientation_attempts=max_orientation_attempts,
+                # TODO: Remaining arguments are unused but required by BasePolicy.
+                #       These will be removed when PositioningProcedure is split from
+                #       BasePolicy
+                #
+                # Note that if we use rng=self.rng below, then the following test will
+                # fail:
+                #   tests/unit/evidence_lm_test.py::EvidenceLMTest::test_two_lm_heterarchy_experiment  # noqa: E501
+                # The test result seems to be coupled to the random seed and the
+                # specific sequence of rng calls (rng is called once on GetGoodView
+                # initialization).
+                rng=np.random.RandomState(),
+                action_sampler_args=dict(actions=[LookUp]),
+                action_sampler_class=UniformlyDistributedSampler,
+                switch_frequency=0.0,
+            )
+            result = self.motor_system._policy.positioning_call(
+                self._observation, self.motor_system._state
+            )
+            while not result.terminated and not result.truncated:
```

Comment:
We could apply De Morgan's law here. I feel like this reads better, but it could just be me.
```suggestion
            while not (result.terminated or result.truncated):
```

Reply:
I'm split on this. Right now, there is one for and one against. If someone else prefers this, I'll switch.